microsoft/testsuites/vm_extensions/azsecpack.py

```diff
@@ -0,0 +1,287 @@
+    @TestCaseMetadata(
+        description="""
+            Verify whether Azure security pack can be installed, uninstalled
+            successfully, and check if the autoconfig is configured successfully.
+        """,
+        priority=0,
+        use_new_environment=True,
+        requirement=simple_requirement(
+            supported_features=[AzureExtension], unsupported_os=[BSD]
+        ),
+    )
+    def verify_azsecpack(self, node: Node, log: Logger, result: TestResult) -> None:
+        environment = result.environment
+        assert environment, "fail to get environment from testresult"
+        platform = environment.platform
+        assert isinstance(platform, AzurePlatform)
+        rm_client = platform._rm_client
+        assert rm_client
+        msi_client = platform._msi_client
+        assert msi_client
+
+        node_context = get_node_context(node)
+        resource_group_name = node_context.resource_group_name
+        location = node_context.location
+        vm_name = node_context.vm_name
+
+        # Add resource tag for AzSecPack
+        tag = {"azsecpack": "nonprod"}
+        add_tag_for_vm(platform, resource_group_name, vm_name, tag, log)
+
+        # Create an AzSecPackAutoConfigRG resource group with the specific location
+        # For Public cloud, the specific location is "eastus".
+        autoconfig_rg_name = "AzSecPackAutoConfigRG"
+        autoconfig_rg_exists = rm_client.resource_groups.check_existence(
+            autoconfig_rg_name
+        )
+        if not autoconfig_rg_exists:
+            params = {"location": "eastus"}
+            rm_client.resource_groups.create_or_update(autoconfig_rg_name, params)
+            log.info(f"{autoconfig_rg_name} is created successfully")
+        else:
+            log.info(f"{autoconfig_rg_name} is already existed")
+
+        # Create a user assigned managed identity
+        msi_name = f"AzSecPackAutoConfigUA-{location}"
+        msi = msi_client.user_assigned_identities.create_or_update(
+            resource_group_name=autoconfig_rg_name,
+            resource_name=msi_name,
+            parameters={"location": location},
+        )
+        log.info(f"{msi.id} is created successfully")
+
+        # Assign the user assigned managed identity to the VM
+        add_user_assign_identity(platform, resource_group_name, vm_name, msi.id, log)
+
+        # Install agent extension
+        self._install_monitor_agent_extension(node)
+        self._install_security_agent_extension(node)
+        self._install_mdsd(node, log)
+
+        # Check and verify
+        self._check_mdsd_service_status(node, log)
+        self._check_azsec_services_status(node, log)
+        self._check_azsecd_status(node, log)
+        self._check_journalctl_logs(node, log)
+        self._check_mdsd_qos_logs(node, log)
+
+        # Clean up the environment
+        if not autoconfig_rg_exists:
+            rm_client.resource_groups.begin_delete(autoconfig_rg_name)
```

Comment:
If it's created with the specified name, don't delete it. Because it may cause conflict, when multiple test runs use it at the same time.

Reply:
Done